hw07

zuwarka opened the pull request.

zuwarka committed:
Create main.rb

zuwarka committed:
Create string_editor.rb

zuwarka committed:
Create indifferent_accessible.rb

Файл с "чудесами"

zuwarka committed:
Create README.md

zuwarka committed:
Create main.rb

zuwarka committed:
Create proc_add_num.rb

zuwarka committed:
Create module_triable.rb

zuwarka committed:
Create README.md

Alex6Alex commented on homework08 (01.08.2022)/proc_add_num.rb:
```diff
@@ -0,0 +1,7 @@
+# который можно использовать для добавления переданного числа к другим числовым значениям.
+
+def add_num(num)
+  proc {|element| element + num}
```
Тут можно сделать отступы от фигурных скобочек: `{ |element| element + num }`

Alex6Alex commented on homework08 (01.08.2022)/module_triable.rb:
```diff
@@ -0,0 +1,17 @@
+module Triable
+  def try
+    yield self if block_given?
+    rescue NoMethodError
```
Тут, пожалуй, стоит явно указать, что будет nil возвращен, часто на это будет ругаться rubocop:

```ruby
rescue .....
  nil
end
```

Alex6Alex commented on homework08 (01.08.2022)/module_triable.rb:
```diff
@@ -0,0 +1,17 @@
+
+module Triable
+  def try
+    yield self if block_given?
```
Хорошо, что есть проверка на block_given. Но тогда стоит немного подправить метод, чтобы он возвращал сам объект, для которого был вызван метод, если блок не был передан. Так, на мой взгляд, метопы был бы более дружественным:
```ruby
1.try #=> 1
{ a: 1 }.try #=> { a: 1 }
```

zuwarka committed:
Update proc_add_num.rb

zuwarka committed:
Update module_triable.rb

Добавлено явное указание nil при возникновении ошибки. В случае отсутствия блока метод возвращает сам объект, вызвавший метод.

zuwarka committed:
Update indifferent_accessible.rb

Добавлен лаконичный работающий вариант

zuwarka committed:
Update README.md

zuwarka committed:
Update README.md
